src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Services/OpenApiDocumentService/OpenApiDocumentServiceTests.Parameters.cs

```diff
@@ -190,4 +191,29 @@ await VerifyOpenApiDocument(builder, document =>
             Assert.Null(document.Paths["/api/content-type-lower"].Operations[OperationType.Get].Parameters);
         });
     }
+
+    [Fact]
+    public async Task GetOpenApiParameters_ToleratesCustomBindingSource()
+    {
+        var action = CreateActionDescriptor(nameof(ActionWithCustomBinder));
+
+        await VerifyOpenApiDocument(action, document =>
+        {
+            var operation = document.Paths["/custom-binding"].Operations[OperationType.Get];
+            var parameter = Assert.Single(operation.Parameters);
+            Assert.Equal("model", parameter.Name);
+            Assert.Null(parameter.In);
```

Comment:
Would we want to/be able to inject a transformer for MVC that takes BindingSource into account?

Reply:
We handle the BindingSource's that are built-in and mappable to `ParameterLocation`s right now. If someone was using a custom binding source, like the Cookie scenario referenced in the issue above is possible with a custom operation transformer. Something like:

```
options.AddOperationTransformer((operation, context, ct) =>
{
   if (context.Description.ParameterDescriptions.Any(p => p.BindingSource == MyCustomBindingSource))
   {
     // Get the ParameterDescription name and lookup into the OpenApiParameter list and set parameter location there
   }
});
```

Reply:
Yeah, the cookie source specifically is the one where we lack parity between [OpenApiParameterLocation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.openapi.models.parameterlocation) and the [BindingSource](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.bindingsource?view=aspnetcore-8.0#fields).